cli/cmd/upgrade.go

```diff
@@ -246,7 +253,7 @@ func (options *upgradeOptions) validateAndBuild(stage string, k kubernetes.Inter
 		}
 		configs.GetGlobal().IdentityContext = toIdentityContext(identity)
 	} else {
-		identity, err = fetchIdentityValues(k, idctx)
+		identity, err = fetchIdentityValues(k, idctx, isIssuerExternal(flags))
```

Comment:
Is this configuration loaded from the linkerd config?  i.e. are the flags populated from the config map? 

This question might be irrelevant if we can get rid of this boolean as per my other comment.

Reply:
Yes, they are coming from the config map